<!-- -->
<launch>
	<arg name="steer_mode" 					default="0" /> <!-- steer angle = 0,  steer torque = 1 -->
	<arg name="drive_mode" 					default="1" /> <!-- linear velocity = 0, pedals stroke = 1 -->
	<arg name="auto_calibration_mode"		default="false" />
	
	<arg name="lowpass_steer_cutoff" 		default="5.0"/> <!-- Hz -->
	<arg name="control_frequency" 			default="50"/> <!-- Hz -->
	
	<arg name="max_steer_value"				default="660"/> <!-- degrees -->	
	<arg name="max_steer_torque"			default="100"/>
	<arg name="min_steer_torque"			default="-100"/> 
	
	<arg name="max_accel_value"				default="100"/>
	<arg name="max_brake_value"				default="100"/>
	
	<arg name="steerGainKP" 				default="30.0"/>
	<arg name="steerGainKI" 				default="5.0"/>
	<arg name="steerGainKD" 				default="2.0"/>
	
	<arg name="velocityGainKP" 				default="1.0"/>
	<arg name="velocityGainKI" 				default="0.0"/>
	<arg name="velocityGainKD" 				default="0.0"/>
	
	<arg name="accelGainKP" 				default="5.0"/>
	<arg name="accelGainKI" 				default="3.0"/>
	<arg name="accelGainKD" 				default="0.005"/>
	
	<arg name="brakeGainKP" 				default="2.0"/>
	<arg name="brakeGainKI" 				default="0.05"/>
	<arg name="brakeGainKD" 				default="0.01"/> 		 	
	
	<node pkg="op_pid_controller" type="op_pid_controller" name="op_pid_controller" output="screen">			
		<param name="steer_mode" 				value="$(arg steer_mode)" />
		<param name="drive_mode" 				value="$(arg drive_mode)" />
		<param name="auto_calibration_mode" 	value="$(arg auto_calibration_mode)" />
		
		<param name="lowpass_steer_cutoff" 		value="$(arg lowpass_steer_cutoff)" />
		<param name="control_frequency" 		value="$(arg control_frequency)" />
		
		<param name="max_steer_value" 			value="$(arg max_steer_value)" />		
		<param name="max_accel_value" 			value="$(arg max_accel_value)" />
		<param name="max_brake_value" 			value="$(arg max_brake_value)" />
				
		<param name="steerGainKP" 				value="$(arg steerGainKP)" />
		<param name="steerGainKI" 				value="$(arg steerGainKI)" />
		<param name="steerGainKD" 				value="$(arg steerGainKD)" />
		
		<param name="velocityGainKP" 			value="$(arg velocityGainKP)" />
		<param name="velocityGainKI" 			value="$(arg velocityGainKI)" />
		<param name="velocityGainKD" 			value="$(arg velocityGainKD)" />
		
		<param name="accelGainKP" 				value="$(arg accelGainKP)" />
		<param name="accelGainKI" 				value="$(arg accelGainKI)" />		
		<param name="accelGainKD" 				value="$(arg accelGainKD)" />
		
		<param name="brakeGainKP" 				value="$(arg brakeGainKP)" />
		<param name="brakeGainKI" 				value="$(arg brakeGainKI)" />
		<param name="brakeGainKD" 				value="$(arg brakeGainKD)" />			
	</node>

</launch>
